conans/client/build/compiler_flags.py

```diff
@@ -0,0 +1,149 @@
+    if str(compiler) in ['clang', 'apple-clang']:
+        if str(libcxx) in ['libstdc++', 'libstdc++11']:
+            flags.cxxflags.append('-stdlib=libstdc++')
+        elif str(libcxx) == 'libc++':
+            flags.cxxflags.append('-stdlib=libc++')
+    elif str(compiler) == 'sun-cc':
+        flags.cxxflags.append({"libCstd": "-library=Cstd",
+                               "libstdcxx": "-library=stdcxx4",
+                               "libstlport": "-library=stlport4",
+                               "libstdc++": "-library=stdcpp"}.get(libcxx, None))
+    return flags
+
+
+def pic_flags(compiler=None):
+    """
+    returns PIC (position independent code) flags, such as -fPIC
+    """
+    if compiler == 'Visual Studio':
+        return CompilerFlags()
+    return CompilerFlags(cflags=['-fPIC'], cxxflags=['-fPIC'])
+
+
+def build_type_flags(compiler=None, build_type=None):
+    """
+    returns flags specific to the build type (Debug, Release, etc.)
+    usually definitions like DEBUG, _DEBUG, NDEBUG, or flags to
+    control debug information (-s, -g, /Zi, etc.)
+    """
+    flags = CompilerFlags()
+    if str(compiler) == 'Visual Studio':
+        if build_type == 'Debug':
+            flags.cflags.append('/Zi')
+            flags.cxxflags.append('/Zi')
+    else:
+        if build_type == 'Debug':
+            flags.cflags.append('-g')
+            flags.cxxflags.append('-g')
+        elif build_type == 'Release' and str(compiler) == 'gcc':
+            flags.cflags.append('-s')
+            flags.cxxflags.append('-s')
+            # TODO: why NDEBUG only for GCC?
+            flags.defines.append('NDEBUG')
+    return flags
+
+
+def adjust_path(path, win_bash=False, subsystem=None, compiler=None):
+    """
+    adjusts path to be safely passed to the compiler command line
+    for Windows bash, ensures path is in format according to the subsystem
+    for path with spaces, places double quotes around it
+    converts slashes to backslashes, or vice versa
+    """
+    if str(compiler) == 'Visual Studio':
+        path = path.replace('/', '\\')
+    else:
+        path = path.replace('\\', '/')
+    if win_bash:
+        path = unix_path(path, subsystem)
+    return '"%s"' % path if ' ' in path else path
+
+
+def sysroot_flags(sysroot, win_bash=False, subsystem=None, compiler=None):
+    if str(compiler) != 'Visual Studio' and sysroot:
+        sysroot = adjust_path(sysroot, win_bash=win_bash, subsystem=subsystem, compiler=compiler)
+        sysroot_flag = ['--sysroot=%s' % sysroot]
+        return CompilerFlags(cflags=sysroot_flag, cxxflags=sysroot_flag, ldflags=sysroot_flag)
+    return CompilerFlags()
+
+
+def format_defines(defines):
+    return ["-D%s" % define for define in defines]
+
+
+def format_include_paths(include_paths, win_bash=False, subsystem=None, compiler=None):
+    return ["-I%s" % adjust_path(include_path, win_bash=win_bash, subsystem=subsystem, compiler=compiler)
```

Comment:
Why -I and not /I with visual?

Reply:
Visual Studio accepts both forms -l and /l, -LIBPATH and /LIBPATH, it doesn't really matter a lot., we can use any of two options. change or keep?

Reply:
Keep it. Np. I will open a PR for you the later on Monday. Thanks for your quick answer